tox.ini

```diff
@@ -7,17 +7,9 @@ envlist =
 requires =
     setuptools >= 30.3.0
     pip >= 19.3.1
-    tox-pypi-filter >= 0.12
-isolated_build = true
 
 [testenv]
 
-# The following option combined with the use of the tox-pypi-filter above allows
-# project-wide pinning of dependencies, e.g. if new versions of pytest do not
-# work correctly with pytest-astropy plugins. Most of the time the pinnings file
-# should be empty.
-pypi_filter = https://raw.githubusercontent.com/astropy/ci-helpers/main/pip_pinnings.txt
```

Comment:
We need to decide if we want to keep this working or not. It was one of the things that people got animated about not loosing in the APE 17 refactor (being able to exclude packages from CI builds for all projects in a single place) but it seems we aren't actually using it any more? Should we drop it or revive it?

Reply:
(SunPy is using it, so I will be updating the plugin anyway)

Reply:
Why? The file is empty.

Reply:
What used to happen in the *before times* was that when a problematic package i.e. pytest or some other thing was released that broke *all CI everywhere* was that it would get pinned out in ci-helpers. This means that all the packages using ci-helpers ended up not installing that problematic package and we could fix all those packages CIs with one commit.

When we transitioned away from ci-helpers to to tox with the APE 17 work, there was a strong argument made that this functionality of ci-helpers should be preserved. So Tom R and I (mostly Tom) wrote the `pypi-filter` plugin which does the same thing. It reads from a pins file, and uses some PyPI proxy magic to make all invocations of pip act like that package version(s) doesn't exist.

What I think has happened is that we have all collectively forgotten this functionality exists and just stopped using it.

SunPy still uses it, we have a pins file in the core repo, and all our "coordinated" packages [use the pins file in the core repo](https://github.com/sunpy/ndcube/blob/main/tox.ini#L13) so if we pin out a troublesome package in the core repo, then all the other packages also ignore it.

Reply:
I can see pros and cons. My projects are spread out across at least 2 orgs, and a pin that breaks one might not matter for another. I prefer to deal with them on a case by case basis. So for me, this is not useful.

So, the question is... is this tool useful for Astropy?

Reply:
Well the kind of pin that would be useful is tox<4 but that won't work here I think 😆

Reply:
There are plenty of other thing that happen often enough which aren't tox 😉

Reply:
@Cadair, @astrofrog any ideas when a rewrite of the `pypi_filter` plugin might be done?